File: src/builder/Builder.java
package builder;

/**
 * @author Sergey Chistyakov <serge.chistyakov@gmail.com>
 *         21.06.2017
 */
@SuppressWarnings("ALL")
public class Builder {
    private int servingSize; // required
    private int servings; // required

    private int calories = 0;
    private int fat = 0;
    private int sodium = 0;
    private int carbohydrate = 0;

    public Builder setServingSize(int servingSize) {
        this.servingSize = servingSize;
        return this;
    }

    public Builder setServings(int servings) {
        this.servings = servings;
        return this;
    }

    public Builder setCalories(int calories) {
        this.calories = calories;
        return this;
    }

    public Builder setFat(int fat) {
        this.fat = fat;
        return this;
    }

    public Builder setSodium(int sodium) {
        this.sodium = sodium;
        return this;
    }

    public Builder setCarbohydrate(int carbohydrate) {
        this.carbohydrate = carbohydrate;
        return this;
    }

    NutritionFacts build() {
        return new NutritionFacts(servingSize, servings, calories, fat, sodium, carbohydrate);
    }

    public static void main(String[] args) {
        NutritionFacts facts = new Builder()
                .setServingSize(100)
                .setServings(2)
                .setCalories(200)
                .setCarbohydrate(80)
                .setFat(25)
                .setSodium(12)
                .build();
    }
}

File: src/builder/Builder.kt
package builder

/**
 * @author Sergey Chistyakov <serge.chistyakov@gmail.com>
 *         21.06.2017
 */
fun main(args: Array<String>) {
  // If our object is mutable.
  val jFacts = JNutritionFacts().apply {
    servingSize = 100
    servings = 2
    calories = 200
    fat = 80
    sodium = 25
    carbohydrate = 12
  }

  // If object is immutable.
  val kFacts = KNutritionFacts(
      servingSize = 100,
      servings = 2,
      calories = 200,
      fat = 80,
      sodium = 25,
      carbohydrate = 12
  )
}
